app/scripts/controllers/registration.js

```diff
@@ -69,19 +69,24 @@ angular.module('confRegistrationWebApp')
       $location.path('/' + $rootScope.registerMode + '/' + conference.id + '/page/' + $scope.conference.registrationPages[0].id);
     }
 
-    $scope.validateAndGoToNext = function (isValid) {
-      if (isValid) {
-        if (angular.isDefined($scope.nextPage)) {
-          $location.path('/' + $rootScope.registerMode + '/' + conference.id + '/page/' + $scope.nextPage.id);
-        } else {
-          $location.path('/reviewRegistration/' + conference.id).search('regType', null);
-        }
+    //setup visited flags array to store visits by a specific registrant to a specific page
+    if(!$rootScope.visitedPages){
+      $rootScope.visitedPages = [];
```

Comment:
Is it ok to use the `$rootScope` for this? I couldn't find a component to put it in that didn't get reloaded when moving between pages.